### PR TITLE
Add Helm operator

### PR DESCRIPTION
**Pull request by cniackz**
### Objective:

To add `Helm Operator` for DirectPV

### Source:

* https://github.com/cniackz/directpv-helm-operator

### Testing Notes:

* https://github.com/cniackz/public/wiki/How-to-test-DirectPV-Operator-%E2%80%90-2

### Needed:

* https://github.com/minio/directpv/pull/821

### CODE REVIEW

**Review by balamurugana — COMMENTED** [review]
You could use helm from https://github.com/minio/directpv/tree/master/helm

**Comment by cniackz** [comment]
Sure @balamurugana I will be using `helm` from https://github.com/minio/directpv/tree/master/helm but also I need to test a little bit more prior the review. I will let you know when I am ready for the review.

**Comment by cniackz** [comment]
Two things:

1. Using the helm chart added previously, just moved from location.
2. Using latest tag 4.0.7 for the current Operator
3. Tested truly in k8s to deploy operator and install directpv

I think this is ready for another round of reviews.

File: helm/Makefile
Comment (balamurugana) [inline helm/Makefile:32]: We would need to use `quay.io/minio/directpv-operator`

File: helm/Makefile
Comment (balamurugana) [inline helm/Makefile:1]: Can we have a bash script to generate operator bundle scripts and patch them with appropriate values for VERSION, IMAGE_TAG_BASE etc?

File: helm/PROJECT
Comment (balamurugana) [inline helm/PROJECT:18]: Could we style this all lowercase `directpvchart` or pretty name `DirectPVChart`?

File: helm/README.md
Comment (balamurugana) [inline helm/README.md:1]: We could remove this file or have a minimal documentation.

File: helm/my-directpv.yaml
Comment (balamurugana) [inline helm/my-directpv.yaml:1]: can we move this to `examples` directory or put it into documentation?

File: helm/pvc.yaml
Comment (balamurugana) [inline helm/pvc.yaml:1]: can we move this to `examples` directory or put it into documentation?

File: helm/Makefile
Comment (cniackz) [reply helm/Makefile:32]: I don't have access to push this image here. Who can help me? I will ask Harsha

File: helm/Makefile
Comment (cniackz) [reply helm/Makefile:1]: Hello @balamurugana, I have added the script `helm/generate.sh` 
Please check!.

**Comment by cniackz** [comment]
Waiting for tests to pass, then I will ask another round of reviews.

File: helm/generate.sh
Comment (balamurugana) [inline helm/generate.sh:51]: This command is required for release privilege. We can only keep other steps here.

File: helm/generate.sh
Comment (balamurugana) [inline helm/generate.sh:63]: This command is required for release privilege. We can only keep other steps here.

File: helm/generate.sh
Comment (balamurugana) [inline helm/generate.sh:55]: Can we use relative path as `./helm/config/manager` ?

File: helm/generate.sh
Comment (cniackz) [reply helm/generate.sh:51]: Hello @balamurugana what do you mean by `keep other steps here`, do you want me to remove this line from the script?, what is the release privilege?, what do you mean by that?. I just want to understand better your comment to proceed accordingly.

Comment (cniackz) [reply helm/generate.sh:51]: Sorry I got it now, will remove it then 👍

File: helm/generate.sh
Comment (cniackz) [reply helm/generate.sh:56]: Oh my bad, I was testing this in my laptop and totally forgot to use relative path. I am sorry about that, I will fix it!.

**Comment by cniackz** [comment]
Waiting for tests to pass, then I will ask another round of reviews.

**Review by balamurugana — DISMISSED** [review]

File: helm/examples/pvc.yaml
Comment (Praveenrajmani) [inline helm/examples/pvc.yaml:7]: Do we need this example? If so, can we use https://github.com/minio/directpv/blob/master/functests/sleep.yaml instead?

File: helm/helm.sh
Comment (cniackz) [reply helm/helm.sh:1]: Thank you @Praveenrajmani for your review!

**Comment by cniackz** [comment]
Waiting for tests to pass, then I ask for another review.

**Comment by cniackz** [comment]
Failures are not related to this PR, look: https://github.com/minio/directpv/pull/828

**Review by balamurugana — APPROVED** [review]

**Review by Praveenrajmani — APPROVED** [review]

**Pull request merged**
